server/src/pages/shot/text-tool.js

```diff
@@ -0,0 +1,260 @@
+      <Localized id="annotationTextSize">
+        <select className={`text-select`} title="Text Size" onChange={this.onChangeTextSize.bind(this)} value={this.state.textSize}>
+          <Localized id="textSizeSmall"><option value="small-text">Small</option></Localized>
+          <Localized id="textSizeMedium"><option value="medium-text">Medium</option></Localized>
+          <Localized id="textSizeLarge"><option value="large-text">Large</option></Localized>
+        </select>
+      </Localized>
+      <ColorPicker activeTool={this.state.tool}
+        setColorCallback={this.setColor.bind(this)}
+        color={this.state.color} />
+      <span className="annotation-divider"></span>
+      <Localized id="annotationTextConfirmButton">
+        <button className={`button transparent confirm-text`} id="confirm-text" onClick={this.onClickConfirm.bind(this)} title="Confirm Text">Confirm</button>
+      </Localized>
+      <Localized id="annotationTextCancelButton">
+        <button className={`button transparent cancel-text`} id="cancel-text" onClick={this.onClickCancel.bind(this)} title="Cancel Text">Cancel</button>
+      </Localized>
+    </div></div>;
+  }
+
+  setColor(color, colorName) {
+    this.setState({color, colorName});
+    this.textInput.current.focus();
+  }
+
+  onClickConfirm(e) {
+    // Exit if user doesn't enter any text
+    if (!this.textInput.current.textContent) {
+       if (this.props.cancelTextHandler) {
+          this.props.cancelTextHandler();
+        }
+        sendEvent("cancel-text", "text-toolbar");
+        return;
+    }
+    const styles = window.getComputedStyle(this.textInput.current);
+
+    const BKG_COLOR = styles["background-color"];
+    const FONT_COLOR = styles.color;
+    const FONT_SIZE = parseInt(styles["font-size"], 10);
+    const x = this.state.left + parseFloat(styles["padding-left"]);
+
+    // For large and small font increase y position by BUFFER_VERTICAL pixel
+    // to fix vertical shift seen when rendering text on canvas
+    let BUFFER_VERTICAL = 0;
+
+    if (FONT_SIZE === 36) {
+      BUFFER_VERTICAL = 1;
+    } else if (FONT_SIZE === 16) {
+      BUFFER_VERTICAL = 0.5;
+    }
+
+    const y = this.state.top + BUFFER_VERTICAL + parseFloat(styles.height) / 2;
+
+    const textCanvas = document.createElement("canvas");
+    textCanvas.width = this.props.baseCanvas.width;
+    textCanvas.height = this.props.baseCanvas.height;
+    const drawingContext = textCanvas.getContext("2d");
+
+    drawingContext.scale(this.props.canvasPixelRatio, this.props.canvasPixelRatio);
+    drawingContext.textBaseline = "middle";
+    drawingContext.fillStyle = `${BKG_COLOR}`;
+    drawingContext.fillRect(this.state.left,
+                            this.state.top,
+                            this.textInput.current.clientWidth,
+                            this.textInput.current.clientHeight);
+    drawingContext.fillStyle = `${FONT_COLOR}`;
```

Comment:
Same here.  The interpolation is unnecessary, and so is assigning into `FONT_COLOR`.